crypto/fipsmodule/rsa/rsa.c

```diff
@@ -704,7 +704,17 @@ int RSA_check_key(const RSA *key) {
     return 1;
   }
 
-  if (!rsa_check_public_key(key)) {
+  const rsa_key_type_t key_type = rsa_detect_key_type(key);
+  if (key_type == RSA_KEY_UNKNOWN) {
+    OPENSSL_PUT_ERROR(RSA, RSA_R_UNKOWN_KEY_TYPE);
+    return 0;
+  }
+
+  if (key_type == RSA_KEY_PUBLIC) {
+    return rsa_check_public_key(key);
```

Comment:
Why do we short-circuit before calling `rsa_check_key_common`? (Should that call come before anything else as it is "common"?

Reply:
this is because, after the refactor, `rsa_check_public_key` [calls](https://github.com/awslabs/aws-lc/pull/436/files#diff-25b26f92c168808944b0db449431b2f1f8bc5cb61b99a9a7aedce57811512f48R144) `rsa_check_key_common` and we want to avoid calling it twice.

why call `rsa_check_key_common` in `rsa_check_public_key` if we already call it in `RSA_check_key`? because there are [other callers](https://github.com/awslabs/aws-lc/search?q=rsa_check_public_key) outside of `RSA_check_key` and we want them to benefit from the validation around `n` and `e` as well.